### PR TITLE
CORE-1674 Folders API documentation

### PR DESCRIPTION
**Pull request by tombusby**
This change is [Reviewable](https://reviewable.io/reviews/scrive/scrive-apidocs/74)

### CODE REVIEW

File: documentation/api/folders/updatefolder.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+---
```
Comment (claasahl) [inline documentation/api/folders/updatefolder.yaml]: Unrelated to this documentation, I noticed that I could still `get`, `update` and `delete` already deleted folders

File: documentation/api/folders/listdocuments.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    Other than the JSON fields shown above, which wrap the output data in a list
+    and show a little metadata, the document JSON for each item in `"documents"`
+    is structured the same as it would be for endpoints that display a given
+    document. _(See the [Document Definition](#document-3) documentation for
+    details)._
+
+  parameters:
+    - name: folder_id
+      in: url
+      type: integer
```
Comment (claasahl) [inline documentation/api/folders/listdocuments.yaml]: the ID in the `Folder`-type is a `string`

```suggestion
      type: string
```

File: documentation/scrive_api.yaml [hunk 3]
```diff
@@ -1039,6 +1039,23 @@ paths:
...
+  /folders/create:
+    $ref: ./api/folders/createfolder.yaml
+
+  /folders/{folder_id}/update:
+    $ref: ./api/folders/updatefolder.yaml
+
+  /folders/{folder_id}/delete:
+    $ref: ./api/folders/deletefolder.yaml
+
+  /folders/{folder_id}/list:
```
Comment (claasahl) [inline documentation/scrive_api.yaml]: a HTTP GET `/api/v2/folders/4380/list` returns a 404

Comment (tombusby) [reply documentation/scrive_api.yaml]: Fix merged to master and due to go live with release 90

File: documentation/api/folders/updatefolder.yaml (hunk 1 above)
Comment (tombusby) [reply documentation/api/folders/updatefolder.yaml]: That's very weird, I'll investigate that now

Comment (tombusby) [reply documentation/api/folders/updatefolder.yaml]: Fix submitted and in review (see CORE-1708).

File: documentation/api/folders/createfolder.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      in: formData
+      type: string
+      format: application/json
+      description: |
+        JSON object containing details of the folder to be created.
+      example: { "name": "new child folder of 1", "parent_id": "1" }
+
+  responses:
+    200:
+      $ref: ../../scrive_api.yaml#/definitions/Folder
```
[the remark this answers is not in the document]
Comment (tombusby) [reply documentation/api/folders/createfolder.yaml]: Yep. I've just confirmed that this is the case. Will rectify. The convention for the other account structure APIs is that create/update then a `GET` should produce that same output twice.

File: documentation/api/folders/updatefolder.yaml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    field, it must be done via `parent_id`*.
+
+  parameters:
+    - name: folder_id
+      in: formData
+      type: string
+      format: application/json
+      description: |
+        JSON object containing details of the folder to be updated.
+      example: { "name": "new child folder of 1", "parent_id": "1" }
```
[the remark this answers is not in the document]
Comment (tombusby) [reply documentation/api/folders/updatefolder.yaml]: Good spot, thanks.

File: documentation/api/folders/createfolder.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (tombusby) [reply documentation/api/folders/createfolder.yaml]: Submitted a PR to fix this issue. See CORE-1709.

**Comment by tombusby** [comment]
@claasahl I've resolved your comments and am ready for the next round of review

File: documentation/api/folders/updatefolder.yaml [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    A full Folder JSON response is returned which displays the state _after_
+    the update operation has been performed. Performing `/update` and then
+    using the `GET` endpoint should produce the same output both times.
+
+    **Important Note**: *You cannot move Folders by updating the `children`
+    field, it must be done via `parent_id`*.
+
+  parameters:
+    - name: folder_id
+      in: formData
```
Comment (claasahl) [inline documentation/api/folders/updatefolder.yaml]:
```suggestion
      in: url
```

File: documentation/api/folders/updatefolder.yaml [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+      type: string
+      format: application/json
```
Comment (claasahl) [inline documentation/api/folders/updatefolder.yaml]:
```suggestion
```

**Comment by tombusby** [comment]
@claasahl cheers, I merged your suggestions

**Pull request merged**